indexer/src/main/java/au/org/aodn/esindexer/utils/GeometryUtils.java

```diff
@@ -137,4 +217,261 @@ protected static Coordinate[] reverseCoordinates(Coordinate[] coords) {
         }
         return reversed;
     }
+    /**
+     * Create a grid based on the area of the spatial extents. Once we have the grid, we can union the area
+     * @param envelope - An envelope that cover the area of the spatial extents
+     * @param cellSize - How big each cell will be
+     * @return - List of polygon that store the grid
+     */
+    protected static List<Polygon> createGridPolygons(Envelope envelope, double cellSize) {
```

Comment:
since you have `TODO` to plan removing hard-coded coordinate system, this method should have the 3rd param `int srid`? so when creating a geometry factory, can use `GeometryFactory geometryFactory = new GeometryFactory(new PrecisionModel(), srid);`

Reply:
I removed the TODO , in fact it now should always pass in CRS84, the reason is because the findPolygonFrom --> findPolygonsFromEXBoundingPolygonType will read the coordinate ref from XML and do the transform
`````
String projection = polygonType.getSrsName();
MathTransform transform = null;
if(projection != null && !projection.isBlank()) {
    try {
        transform = CRS.findMathTransform(CRS.decode(projection), CRS84);
    }
    catch (FactoryException e) {
        // Default is WGS84 (CRS84), so do nothing need to transform
    }
}
`````

the other function findPolygonsFromEXGeographicBoundingBoxType seems refer to bounding box which seems already construct correctly.